client/allocdir/task_dir.go

```diff
@@ -57,8 +57,10 @@ func newTaskDir(logger *log.Logger, allocDir, taskName string) *TaskDir {
 	}
 }
 
-// Build default directories and permissions in a task directory.
-func (t *TaskDir) Build(chroot map[string]string, fsi cstructs.FSIsolation) error {
+// Build default directories and permissions in a task directory. If the built
+// boolean is true Build assumes the task dir has already been built and skips
+// expensive operations like chroot creation.
+func (t *TaskDir) Build(built bool, chroot map[string]string, fsi cstructs.FSIsolation) error {
```

Comment:
I don't like `built` since you wouldn't call Build if that is true.  Can it be chrootCreated